apps/src/templates/progress/ProgressPill.jsx

```diff
@@ -50,12 +51,18 @@ class ProgressPill extends React.Component {
     }
 
     let url = levels[0].url;
+    let params = {};
 
     if (selectedSectionId) {
-      url += stringifyQueryParams({section_id: selectedSectionId});
+      params['section_id'] = selectedSectionId;
     }
 
-    return url;
+    const userId = queryParams('user_id');
```

Comment:
Hmmm.  I see that ProgressBubbleSet can actually take a `selectedStudentId` and adding a similar property to ProgressPill would be more consistent.  BUT, it looks like `selectedStudentId` is not passed through to ProgressBubbleSet and the parameter is actually eventually added way down in `BubbleFactory.getBubbleUrl`.

Looks like there's more cleanup that should be done here in the future, but maybe we should call `BubbleFactory.getBubbleUrl` here to fix this specific issue?

(Side comment: As we clean things up, we should probably try to look at the query params in fewer places.  Ideally, we would extract it once near the top, set it in the appropriate state, and then rely on the state elsewhere.)

Reply:
You're right, this could be extracted as a further clean-up. Adding a BubbleFactory.getBubbleUrl makes sense to me

Reply:
I took a quick look at this and decided I'm not going to tackle this right now. We're not consistently sending down `selectedSectionId` through props and I think part of the reason that `getBubbleUrl` is returning the right url is because it has a `preserveQueryParams` parameter which is true for the `ProgressBubble`, so even if `studentId` and `sectionId` aren't passed in, it looks like we just grab the existing values from the URL

Reply:
Sounds good, could you open a ticket for this?  I'd like to revisit this at some point, perhaps when we do more work on the script overview page.  Thanks!

Reply:
https://codedotorg.atlassian.net/browse/LP-2180